codalab/apps/chahub/tasks.py

```diff
@@ -0,0 +1,221 @@
+    if not settings.CHAHUB_API_URL:
+        raise ChahubException("CHAHUB_API_URL env var required to send to Chahub")
+    if not settings.CHAHUB_API_KEY:
+        raise ChahubException("No ChaHub API Key provided")
+
+    obj = get_obj(app_label, model_name, pk, include_deleted=True)
+
+    url = "{}{}{}/".format(settings.CHAHUB_API_URL, obj.get_chahub_endpoint(), pk)
+    logger.info("ChaHub :: Sending to ChaHub ({}) delete message".format(url))
+
+    headers = {'X-CHAHUB-API-KEY': settings.CHAHUB_API_KEY}
+
+    try:
+        resp = requests.delete(url=url, headers=headers)
+    except requests.exceptions.RequestException:
+        resp = None
+
+    if resp and resp.status_code == 204:
+        logger.info("ChaHub :: Received response {} {}".format(resp.status_code, resp.content))
+        obj.delete()
+    else:
+        status = getattr(resp, 'status_code', 'N/A')
+        body = getattr(resp, 'content', 'N/A')
+        logger.info("ChaHub :: Error sending to chahub, status={}, body={}".format(status, body))
+        obj.chahub_needs_retry = True
+        obj.save(force_to_chahub=False)
+
+
+def batch_send_to_chahub(model, limit=None, retry_only=False):
+    qs = model.objects.all()
+    if retry_only:
+        qs = qs.filter(chahub_needs_retry=True)
+    if limit is not None:
+        qs = qs[:limit]
+
+    endpoint = model.get_chahub_endpoint()
+    data = [obj.get_chahub_data() for obj in qs if obj.get_chahub_is_valid()]
+    if not data:
+        logger.info('Nothing to send to Chahub at {}'.format(endpoint))
+        return
+    try:
+        logger.info("Sending all data to Chahub at {}".format(endpoint))
+        resp = _send(endpoint=endpoint, data=data)
+        logger.info("Response Status Code: {}".format(resp.status_code))
+        if resp.status_code != 201:
+            logger.warning('ChaHub Response Content: {}'.format(resp.content))
+    except ChahubException:
+        logger.info("There was a problem reaching Chahub. Retry again later")
+
+
+def chahub_is_up():
+    if not settings.CHAHUB_API_URL:
+        return False
+
+    logger.info("Checking whether ChaHub is online before sending retries")
+    try:
+        response = requests.get(settings.CHAHUB_API_URL)
+        if response.ok:
+            logger.info("ChaHub is online")
+            return True
+        else:
+            logger.info("Bad Status from ChaHub")
+            return False
+    except requests.exceptions.RequestException:
+        # This base exception works for HTTP errors, Connection errors, etc.
+        logger.info("Request Exception trying to access ChaHub")
+        return False
+
+
+def get_chahub_models():
+    return inheritors(ChaHubSaveMixin)
+
+
+@task(queue='site-worker')
+def do_chahub_retries(limit=None):
+    if not chahub_is_up():
+        return
+    chahub_models = get_chahub_models()
+    logger.info('Retrying for ChaHub models: {}'.format(chahub_models))
+
+    chahub_models = inheritors(ChaHubSaveMixin)
```

Comment:
use `get_chahub_models`?

Reply:
believe i had to use this way because of python2 and lower versions of Django. in 1.8 I might be able to? would rather not have to confirm that and stick with this, which should be working

Reply:
we were doing `get_chahub_models` above anyway already, I'm getting tired :(